srcCpp03/RTIDDSImpl.cxx

```diff
@@ -1719,61 +1818,178 @@ IMessagingReader *RTIDDSImpl<T>::CreateReader(
         const std::string &topic_name,
         IMessagingCB *callback)
 {
-    using namespace dds::core::policy;
-    using namespace rti::core::policy;
-
     std::string qos_profile;
     qos_profile = get_qos_profile_name(topic_name);
     if (qos_profile.empty()) {
         throw std::logic_error("[Error] Topic name");
     }
 
-    dds::core::QosProvider qos_provider = getQosProviderForProfile(
-            _PM->get<std::string>("qosLibrary"),
-            qos_profile);
-    dds::sub::qos::DataReaderQos dr_qos = qos_provider.datareader_qos();
-
+    dds::sub::qos::DataReaderQos dr_qos = setup_DR_QoS(qos_profile, topic_name);
 
-    Reliability qos_reliability = dr_qos.policy<Reliability>();
-    ResourceLimits qos_resource_limits = dr_qos.policy<ResourceLimits>();
-    Durability qos_durability = dr_qos.policy<Durability>();
-    rti::core::policy::DataReaderProtocol dr_DataReaderProtocol =
-            dr_qos.policy<rti::core::policy::DataReaderProtocol>();
-
-    // This will allow us to load some properties.
-    std::map<std::string, std::string> properties =
-            dr_qos.policy<Property>().get_all();
+    if (!_PM->get<bool>("dynamicData")) {
+        dds::topic::Topic<T> topic(_participant, topic_name);
+        dds::sub::DataReader<T> reader(dds::core::null);
+        ReceiverListener<T> *reader_listener = NULL;
 
-    // only force reliability on throughput/latency topics
-    if (topic_name != ANNOUNCEMENT_TOPIC_NAME.c_str()) {
-        if (!_PM->get<bool>("bestEffort")) {
-            qos_reliability = dds::core::policy::Reliability::Reliable();
+        if (topic_name == THROUGHPUT_TOPIC_NAME.c_str() && _PM->is_set("cft")) {
```

Comment:
This is a pointer comparison, not a string comparison. Is that what you want?

Even if you do, the full string comparison (`topic_name == THROUGHPUT_TOPIC_NAME`) is more robust and maintainable.

Reply:
I moved that code from `CreateReader` function to `setup_DR_QoS` but this comparison was there previous to my commits, maybe @javmorcas knows more about that. 

As far as I understand we should rather not do a pointer comparison there since in case we hardcode the `topic_name` argument when calling `setup_DR_QoS` function, this comparison will not work as desired. I agree that string comparison is a better approach.

 I will change this and other similar cases on my next commit.